client/data/plugins/use-core-plugins-query.ts

```diff
@@ -24,17 +24,27 @@ export type CorePluginsResponse = CorePlugin[];
 
 /**
  * Fetches the plugins for a given site
- * @param siteIdOrSlug The site ID or slug
  */
 export const useCorePluginsQuery = (
 	siteIdOrSlug: SiteId | SiteSlug | undefined,
-	hideManagedPlugins: boolean = false
+	hideManagedPlugins = false,
+	addPluginExtension = false
 ): UseQueryResult< CorePluginsResponse > => {
 	const select = ( plugins: CorePluginsResponse ) => {
-		const decodedPlugins = decodeEntitiesFromPlugins( plugins );
-		return hideManagedPlugins
-			? decodedPlugins.filter( ( plugin ) => ! plugin.is_managed )
-			: decodedPlugins;
+		let _plugins = decodeEntitiesFromPlugins( plugins );
+
+		if ( hideManagedPlugins ) {
+			_plugins = _plugins.filter( ( plugin ) => ! plugin.is_managed );
+		}
+
+		if ( addPluginExtension ) {
+			_plugins = _plugins.map( ( plugin ) => ( {
+				...plugin,
+				plugin: `${ plugin.plugin }.php`,
```

Comment:
Not a blocker and I think this probably won't happen but should we be extra safe and check if the extension is already existed before appending it?

Reply:
Good suggestion. I'll add it. 👌